### PR TITLE
[JENKINS-60088] set label "jenkins/label=<label>" instead of "jenkins/<label>=true"

### PR DESCRIPTION
**Pull request by MattLud**
[JENKINS-60088](https://issues.jenkins-ci.org/browse/JENKINS-60088)

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 1]
```diff
@@ -381,7 +381,7 @@ public String getActiveDeadlineSecondsStr() {
         ImmutableMap.Builder<String, String> builder = ImmutableMap.<String, String> builder();
         if (!labelSet.isEmpty()) {
             for (LabelAtom label : labelSet) {
-                builder.put(label == null ? DEFAULT_ID : "jenkins/" + label.getName(), "true");
+                builder.put("buildlabel",label == null ? DEFAULT_ID : "jenkins_" + label.getName());
```
Comment (Vlatombe) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Why not `jenkins/label` as a key?

Comment (MattLud) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: No reason it can't be that - will change it.

**Comment by MattLud** [comment]
Passing now but I want point out this has a breaking change where labels with violating character requirements will fail.

**Comment by Vlatombe** [comment]
Need to sanitize label maybe?

**Comment by MattLud** [comment]
It may not need sanitizing - it uses the build name, which already gets rejected if it fails the naming regex that applies here.

Only reason to sanitize is to meet the 63 character limit which I can add in and provide a test if desired.

**Comment by Vlatombe** [comment]
generated labels should be safe already (https://github.com/jenkinsci/kubernetes-plugin/blob/3f91dcd194a4bbb2b45a21d07869928cec656cb0/src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java#L167)

**Comment by MattLud** [comment]
Moving this out of draft now - feel free to merge.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodLabel.java [hunk 2]
```diff
@@ -112,5 +112,5 @@ public String toString() {
         public String getDisplayName() {
             return "Pod Label";
         }
-    }
+    }
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodLabel.java]: Would you mind reverting unrelated whitespace changes? The patch to `src/main/` should be one line.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java [hunk 3]
```diff
@@ -75,9 +75,9 @@
     private String image;
 
     private boolean privileged;
-    
+
```
Comment (MattLud) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: @jglick  - for these trailing whitespaces, do you want them to stay as well?

Comment (jglick) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: No strong opinion (and I do not want to slow down a potentially important fix like this one). On the one hand we do not want trailing whitespace.¹ On the other hand it is best for PRs of substance to have minimal diffs—easier to review, less likely to have merge conflicts, easier to cherry-pick, less churn in blame view, etc. My general tendency is to err on the side of keeping diffs minimal, and if whitespace or other formatting is bothersome, to make an occasional commit/PR² which _only_ cleans this up and clearly does not affect functionality.

----

¹Best to configure an editor to strip them when pressing **Enter** or whatever—but not to touch lines that were not actually inserted or edited.

²There have been some proposals to set up mandatory code style checks in Jenkins plugins which opt in, so that after a painful baseline commit which touches basically every line of code in the repository, you can forget about the issue going forward; but I do not think this has had much momentum.

Comment (MattLud) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java]: Done and done. Changed editor to only modify whitespace on modified lines.

**Comment by fraz3alpha** [comment]
> Can you provide an example pipeline?

@MattLud 

We have the same issue on all the Jenkins instance where we have upgraded to 1.22.1, our default pod template in the cloud configuration (in Manage / Configure) has two labels, but gives the duplicate label error. We can get it to work by removing all but one of them, e.g. to leave:

Screenshot 2019-12-18 at 17 56 53

Our test pipeline that requires this template only specifies a single label - so it's a trivial test case from the pipeline end, but the template it needs to start will fail to do so if it has more than one label defined.

This has unfortunately meant we have had to remove all the useful matrix labelling in this instance, so we can't leave it like this, and will have to stay on the downgraded plugin until this is fixed and multiple labels are supported again.

**Comment by Vlatombe** [comment]
Filed as https://issues.jenkins-ci.org/browse/JENKINS-60537
